zubhub_frontend/zubhub/src/components/inputText/inputText.jsx

```diff
@@ -0,0 +1,80 @@
+import React, {useState} from 'react';
+import clsx from 'clsx';
+import ReactQuill from 'react-quill';
+import 'react-quill/dist/quill.snow.css';
+import 'react-toastify/dist/ReactToastify.css';
+import { handleInputTextFieldChange, handleInputTextFieldBlur } from '../../views/create_activity/createActivityScripts'
+import {
+  Box,
+  Typography,
+  FormControl,
+  FormHelperText,
+  ClickAwayListener
+} from '@material-ui/core';
+
+function InputText(props) {
+    const { classes, 
+            common_classes,
+            activity_classes, 
+            label, 
+            helperText, 
+            placeholder, 
+            vars
+            } = props 
+    const [inputTextFieldFocused, setInputTextFieldFocused] = useState(false)
+    props = {...props, inputTextFieldFocused, setInputTextFieldFocused}         
+    return (
+      <div>
+          <Typography
+            color="textSecondary"
+            variant="caption"
+            component="span"
+            className={clsx(
+              classes.fieldHelperTextStyle,
+              common_classes.marginBottom1em,
+            )}
+          >
+            {helperText? helperText : ''}
+          </Typography>
+          <FormControl
+            className={clsx(classes.margin, classes.textField)}
+            variant="outlined"
+            size="small"
+            fullWidth
+            margin="none"
+            >
+            <ClickAwayListener
+              onClickAway={() =>
+                {if (inputTextFieldFocused){
+                  console.log('blur event from :', label)
+                  handleInputTextFieldBlur( label, props )
```

Comment:
when creating a function please create an object parameters instead, It fixes three problems
* we don't have pass the params in order
* As the function parameters increase its hard to what we are sending vs what we are consuming in the function
* If we need to add or remove parameters it would be a very simple change

Reply:
And Please try to send only the required params instead of sending all the props